packages/txservice/src/provider.ts

```diff
@@ -461,128 +492,197 @@ export class ChainRpcProvider {
    *
    * @returns A hash string address belonging to the signer.
    */
-  public getAddress(): ResultAsync<string, TransactionError> {
-    return this.resultWrapper<string>(true, async () => {
-      return await this.signer!.getAddress();
-    });
+  public async getAddress(): Promise<string> {
+    this.checkSigner();
+    return await this.signer!.getAddress();
   }
 
   /**
-   * Gets a transaction.
+   * Retrieves a transaction's receipt by the transaction hash.
    *
    * @param hash - the transaction hash to get the receipt for.
    *
    * @returns A TransactionReceipt instance.
    */
-  public getTransactionReceipt(hash: string): ResultAsync<providers.TransactionReceipt, TransactionError> {
-    return this.resultWrapper<providers.TransactionReceipt>(false, async () => {
-      const receipt = await this.provider.getTransactionReceipt(hash);
+  public async getTransactionReceipt(hash: string): Promise<providers.TransactionReceipt> {
+    return this.execute<providers.TransactionReceipt>(false, async (provider: SyncProvider) => {
+      const receipt = await provider.getTransactionReceipt(hash);
       return receipt;
     });
   }
 
   /**
    * Gets the current transaction count.
    *
-   * @param blockTag - the block tag to get the transaction count for. Use "latest" mined-only transactions.
+   * @param blockTag (default: "latest") - The block tag to get the transaction count for. Use "latest" mined-only transactions.
    * Use "pending" for transactions that have not been mined yet, but will (supposedly) be mined in the pending
    * block (essentially, transactions included in the mempool, but this behavior is not consistent).
    *
-   * @returns Number of transactions sent; by default, including transactions in the pending (next) block.
+   * @returns Number of transactions sent AKA the current nonce.
    */
-  public getTransactionCount(blockTag = "pending"): ResultAsync<number, TransactionError> {
-    // If it's been less than a couple seconds since we retrieved tx count, return the cached value.
-    if (this.cachedTransactionCount && Date.now() - this.cachedTransactionCount.timestamp < 2_000) {
-      return okAsync(this.cachedTransactionCount.value);
+  public async getTransactionCount(blockTag = "latest"): Promise<number> {
+    // TODO: Cache both latest and pending transaction counts separately?
+    if (this.cache.data.transactionCount && blockTag === "latest") {
+      return this.cache.data.transactionCount;
     }
 
-    return this.resultWrapper<number>(true, async () => {
-      const value = await this.signer!.getTransactionCount(blockTag);
-      this.cachedTransactionCount = { value, timestamp: Date.now() };
-      return value;
+    return this.execute<number>(true, async (provider: SyncProvider) => {
+      const transactionCount = await provider.getTransactionCount(await this.signer!.getAddress(), blockTag);
+      this.cache.set({ transactionCount });
```

Comment:
how long does this live?

Reply:
2 seconds, see the hardcoded TTL in the constructor lol

we can make this configurable/based on block period later